### PR TITLE
Add a new endpoint template-loader to resolve the correct template by type

### PR DESCRIPTION
**Pull request by arthur791004**
Related to https://github.com/Automattic/wp-calypso/issues/71375, https://github.com/Automattic/wp-calypso/pull/71702

#### Changes proposed in this Pull Request:

* Before we use a tricks `/<page>?_wp-find-template=true` to resolve the template of the current page, but it doesn't work if we want to proxy the request from wpcom to Jetpack due to the authentication issue. Thus, we propose to add a new endpoint called `template-loader` to resolve the template by its type, e.g.: home.

#### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [x] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [x] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

#### Jetpack product discussion

N/A

#### Does this pull request change what data or activity we track or use?

N/A

#### Testing instructions:

* Create a [Jurassic Ninja sandbox](https://jurassic.ninja/create/?jetpack-beta&branch=add/wpcom-v2-template-loader-endpoint) against this PR.
* Setup Jetpack on your site to connect with WordPress
* Apply D97093-code to your sandbox
* Apply this PR to your sandbox
  `bin/jetpack-downloader test jetpack add/wpcom-v2-template-loader-endpoint`
* Go to API Console and follow the steps below to verify you're able to get correct response
  * Select `WP REST API > wpcom/v2`
  * Call `/sites/<your_site/template-loader/home`

### CODE REVIEW

**Review by sergeymitr — DISMISSED** [review]
Tests well, the code looks good.
However, it still requires some changes on the WPCOM side to properly set it up.
It should be safe to merge it now, but it won't work until the WPCOM diff lands that properly proxies it into Jetpack.
Please let me know if I can help with setting it up.

**Comment by arthur791004** [comment]
@sergeymitr Thanks for the review!

> However, it still requires some changes on the WPCOM side to properly set it up.
> It should be safe to merge it now, but it won't work until the WPCOM diff lands that properly proxies it into Jetpack.
> Please let me know if I can help with setting it up.

Yep. If I understand correctly, we still need to land D97093-code on WPCOM after we merge this PR and sync the change to WPCOM (Or do we have to wait until the next jetpack release?), right?

**Comment by arthur791004** [comment]
@sergeymitr I rebased this PR onto trunk and found it needs to be approved again 😅 Could you help with that?

BTW, I'm AFK next week, so feel free to merge it. Thank you!

**Review by jeherve — APPROVED** [review]

**Comment by jeherve** [comment]
> I rebased this PR onto trunk and found it needs to be approved again 😅

I've approved it again, and will merge. In the future, do not hesitate to merge instead of rebase; doing that will not invalidate existing reviews :)

**Comment by arthur791004** [comment]
Okay, thanks for merging this PR 👍

**Pull request merged**
